swift/apple/FirezoneKit/Sources/FirezoneKit/Helpers/LogCompressor.swift

```diff
@@ -9,91 +9,116 @@ import AppleArchive
 import Foundation
 import System
 
-struct LogCompressor {
-  private let destinationURL: URL?
-  let fileName: String
-
-  init(destinationURL: URL? = nil) {
-    let dateFormatter = ISO8601DateFormatter()
-    dateFormatter.formatOptions = [.withFullDate, .withTime, .withTimeZone]
-    let timeStampString = dateFormatter.string(from: Date())
-    self.fileName = "firezone_logs_\(timeStampString).aar"
-    self.destinationURL = destinationURL
+/// This module handles the business work of interacting with the AppleArchive framework to do the actual
+/// compression. It's used from both the app and tunnel process in nearly the same way, save for how the
+/// writeStream is opened.
+///
+/// In the tunnel process, the writeStream is a custom stream derived from our TunnelArchiveByteStream,
+/// which keeps state around the writing of compressed bytes in order to handle sending chunks back to the
+/// app process.
+///
+/// In the app process, the writeStream is derived from a passed file path where the Apple Archive
+/// framework handles writing for us -- no custom byte stream instance is needed.
+///
+/// Once the writeStream is opened, the remaining operations are the same for both.
+public struct LogCompressor {
```

Comment:
This class doesn't have any members, why do we need a class here? Couldn't all of these just be free-functions?

Reply:
Well, for this entity we need to maintain an instance of it in the PacketTunnelProvider to keep it alive across IPC calls.

In practice it's uncommon not to namespace your utility functions with _something_ - I presume it's because Swift's package / module system is fairly basic compared to something like Rust.

Reply:
What state is stored in the instance that we need to retain? Maybe I am not reading this correctly but without fields, what state are you storing?

Reply:
The `LogCompressor` instance state. If we don't store an instance to it, the streams will close after the IPC's `completionHandler` is called, i.e. after the first `flush()` and that will close the `byteStream`

Reply:
So I just test this by only storing a reference to `tunnelArchiveByteStream` instance in the `logExportState` enum, and this causes the export to never complete and just hang.

The reason is because the `LogCompressor` instance gets swept, the streams get closed, and there's nothing calling our `tunnelArchiveByteStream`'s handlers anymore which would trigger the next `flush()` and so forth.

It's a bit of an odd interface.